src/App.tsx

```diff
@@ -15,36 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  length = 'length',
+  alphabet = 'alphabet',
+  default = '',
+}
+
+function getPraparedGoods(goods: string[], sortField: string) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  let visibleGoods = getPraparedGoods(goodsFromServer, sortField);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabet,
+          })}
+          onClick={() => setSortField(SortType.alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+          onClick={() => setSortField(SortType.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField.length > 0 || reversed) && (
```

Comment:
```suggestion
        {(sortField || reversed) && (
```
`sortField` is a string, so just check if it exists. And create a separate variable for this condition to avoid combining different logical operators in the same place